### DISCUSSION TITLE
Corrupted/Future data from Inverter

### DISCUSSION THREAD

**Discussion by alefhar** [SBFspot/SBFspot#577] [Q&A]
Since a few days my inverter seems to be acting up and has old data stored for the current day. The data up to the current 5-minute interval gets overwritten but data that has a timestamp in the future is stale. This causes `./SBFspot -ad1` to read the correct data up to the current time and the corrupted, future data. Everything is then written to the database and uploaded to PVOutput. The values in the database aren't overwritten when new data becomes available.
I have an example here: [pasebin](https://pastebin.com/kU6PXiCv)
There is a very high power value and after that the total energy is lower and power is always zero:
```
21/09/2022 11:40:00 : 6441.234kWh - 456.000W
21/09/2022 11:45:00 : 5672.121kWh - 61489146903135816.000W
21/09/2022 11:50:00 : 5672.121kWh - 0.000W
```
Currently, I've changed my cron jobs to archive day data only once in the evening.

Maybe SBFspot could not accept data with timestamps in the future?

**Comment by sbf-** [comment]
> Maybe SBFspot could not accept data with timestamps in the future?

Sounds like a valid option... or when energy value (kWh) lower than previous one.
Power is calculated from these two values, resulting in a very high number.

I'm curious how SunnyExplorer handles this case

**Reply by alefhar** [reply]
Seems like SunnyExplorer (as well as SunnyPortal) ignores future data.
Screen of the inverter graph:
![Inverter](https://user-images.githubusercontent.com/191115/191901688-bd154fbd-8eb8-4c32-9639-f3be58b4cc24.png)

And of the SunnyExplorer graph:
![SunnyExplorer](https://user-images.githubusercontent.com/191115/191901690-e1ebc7cd-28ac-4bf1-9a91-39f605318c2e.png)
